semgrep/semgrep/evaluation.py

```diff
@@ -166,6 +166,50 @@ def get_comparison_range_matches(
     return result
 
 
+def compare_where_python(where_expression: str, pattern_match: PatternMatch) -> bool:
+    result = False
+    return_var = "semgrep_pattern_return"
+    lines = where_expression.strip().split("\n")
+    to_eval = "\n".join(lines[:-1] + [f"{return_var} = {lines[-1]}"])
+
+    local_vars = {
+        metavar: pattern_match.get_metavariable_value(metavar)
+        for metavar in pattern_match.metavars
+    }
+    scope = {"vars": local_vars}
+
+    try:
+        # fmt: off
+        exec(to_eval, scope)  # nosem: contrib.dlint.dlint-equivalent.insecure-exec-use, python.lang.security.audit.exec-detected.exec-detected
+        # fmt: on
+        result = scope[return_var]  # type: ignore
+    except KeyError as ex:
+        logger.error(
+            f"could not find metavariable {ex} while evaluating where-python expression '{where_expression}', consider case where metavariable is missing"
+        )
+    except Exception as ex:
+        logger.error(
+            f"received error '{repr(ex)}' while evaluating where-python expression '{where_expression}'"
+        )
+
+    if not isinstance(result, bool):
+        raise SemgrepError(
+            f"where-python expression '{where_expression}' needs boolean output but got {result}"
+        )
+    return result
+
+
+def get_where_python_range_matches(
+    where_expression: str, ranges: Set[Range], pattern_matches: List[PatternMatch]
+) -> Set[Range]:
+
```

Comment:
Would be helpful to have function level comment 
```
Returns elements in RANGES where running WHERE_EXPRESSION on an element in PATTERN_MATCHES returns true
```
? or something better worded :P